jscomp/stdlib/hashtbl.js

```diff
@@ -179,25 +174,25 @@ function resize(indexfun, h) {
 
 function key_index(h, key) {
   if (h.length >= 3) {
-    return Caml_primitive.caml_hash(10, 100, h[3], key) & h[2].length - 1;
+    return Caml_primitive.caml_hash(10, 100, h[2], key) & h[1].length - 1;
   }
   else {
-    return Caml_primitive.caml_hash_univ_param(10, 100, key) % h[2].length;
+    return Caml_primitive.caml_hash_univ_param(10, 100, key) % h[1].length;
   }
 }
 
 function add(h, key, info) {
   var i = key_index(h, key);
-  var bucket_003 = h[2][i];
-  var bucket = [
-    /* Cons */0,
-    key,
-    info,
-    bucket_003
-  ];
-  h[2][i] = bucket;
-  ++ h[1];
-  if (h[1] > (h[2].length << 1)) {
+  var bucket_002 = h[1][i];
```

Comment:
we can have a `js_intermediate_flag` to mark special flag of `bucket_002`, if it is used only once, we just inline it later